tests/test_obstacle_map.py

```diff
@@ -0,0 +1,107 @@
+from SDA import *
+import unittest
+import numpy 
+
+class Test_Obstacle_map(unittest.TestCase):
+	"""docstring for test_Obstacle_map"""
+	
+	def setUp(self):
+		self.test_drone_point=numpy.array([0,0,0])
+		self.test_fly_zones=numpy.array([numpy.array([(0,0), (0,100), (100,100), (100,0)])])
+		self.obstacle_map=ObstacleMap(self.test_drone_point,self.test_fly_zones)
+
+	def test_add_obstacle(self):
+
+		# test if the last obstacle in obstacle_map is the added object
+		
+		self.test_Obstacle=StationaryObstacle(numpy.array([1,1,1]),2,1)
+		self.obstacle_map.add_obstacle(self.test_Obstacle)
+		
+		self.assertEqual(self.obstacle_map.get_obstacles(),self.test_Obstacle)
+
+	def test_add_waypoint(self):
+
+		# test if the last waypoint in waypoint_holder is the added way point
+
+		self.new_waypoint=numpy.array([1,1,1])
+		
+		self.obstacle_map.add_waypoint(self.new_waypoint)
+		
+		self.added_waypoint=self.obstacle_map.get_drone().get_waypoint_holder().get_current_waypoint()
+
+		self.assertEqual(self.added_waypoint.all(),self.new_waypoint.all())
+
+	def test_set_drone_position(self):
+
+		# set drone to a new position and check if drone's current position has changed into the new point
+
+		self.new_point=numpy.array([2,2,2])
+
+		self.obstacle_map.set_drone_position(self.new_point)
+
+		self.assertEqual(self.obstacle_map.get_drone().get_point().all(),self.new_point.all())
+
+	def test_reset_obstacles(self):
+
+		# add, reset, then test if there is still any obstcle
+
+		self.test_obstacle=StationaryObstacle(numpy.array([1,1,1]),2,1)
+		self.obstacle_map.add_obstacle(self.test_obstacle)
+		self.obstacle_map.reset_obstacles()
+		self.assertEqual(len(self.obstacle_map.get_obstacles()),0)
+
+	def test_reset_waypoint(self):
+		# add, rest then test if there is any waypoint
+
+		self.test_waypoint=numpy.array([1,1,1])
+		self.obstacle_map.add_waypoint(self.test_waypoint)
+		self.obstacle_map.reset_waypoints()
+		self.assertEqual(self.obstacle_map.get_drone().get_waypoint_holder().__len__(),0)
+
+	def test_is_obstacle_in_path(self):
+
+		# reset the obstacle list first, given a new point and check the return value
+		# according to the current value, the obstacle should be in path so the return value should be true
+
+		self.obstacle_map.get_drone().reset_waypoint()
+		self.obstacle_map.get_drone().add_waypoint(np.array([2,2,2]))
```

Comment:
You need to add an obstacle in this unit test. Because unit tests are not always executed sequentially, the obstacle needs to be created and added in this method